.github/workflows/test-download-artifact-s3-action.yml

```diff
@@ -0,0 +1,118 @@
+      working-directory: '.github/actions/download-artifact-s3'
+
+    - name: Compile
+      run: npm run build
+      working-directory: '.github/actions/download-artifact-s3'
+
+    - name: Lint
+      run: npm run lint
+      working-directory: '.github/actions/download-artifact-s3'
+
+    - name: Format
+      run: npm run format-check
+      working-directory: '.github/actions/download-artifact-s3'
+
+    # Test end-to-end by uploading two artifacts and then downloading them
+    - name: Create artifacts
+      run: |
+        mkdir -p path/to/artifact-A
+        mkdir -p path/to/artifact-B
+        echo "Lorem ipsum dolor sit amet" > path/to/artifact-A/file-A.txt
+        echo "Hello world from file B" > path/to/artifact-B/file-B.txt
+
+    - name: Upload artifact A
+      uses: actions/upload-artifact@v4
+      with:
+        name: Artifact-A-${{ matrix.runs-on }}
+        path: path/to/artifact-A
+
+    - name: Upload artifact B
+      uses: actions/upload-artifact@v4
+      with:
+        name: Artifact-B-${{ matrix.runs-on }}
+        path: path/to/artifact-B
+
+    # Test downloading a single artifact
+    - name: Download artifact A
+      uses: ./.github/actions/download-artifact-s3
+      with:
+        name: 'Artifact-A-${{ matrix.runs-on }}'
+        path: some/new/path
+
+    # Test downloading an artifact using tilde expansion
+    - name: Download artifact A
+      uses: ./.github/actions/download-artifact-s3
+      with:
+        name: 'Artifact-A-${{ matrix.runs-on }}'
+        path: ~/some/path/with/a/tilde
+
+    - name: Verify successful download
+      run: |
+        $file1 = "${{ github.workspace }}/some/new/path/file-A.txt"
+        $file2 = "~/some/path/with/a/tilde/file-A.txt"
+        if(!(Test-Path -path $file1) -or !(Test-Path -path $file2))
+        {
+            Write-Error "Expected files do not exist"
+        }
+        if(!((Get-Content $file1) -ceq "Lorem ipsum dolor sit amet") -or !((Get-Content $file2) -ceq "Lorem ipsum dolor sit amet"))
+        {
+            Write-Error "File contents of downloaded artifacts are incorrect"
+        }
+      shell: pwsh
+
+    # Test downloading both artifacts at once
+    - name: Download all Artifacts
+      uses: ./.github/actions/download-artifact-s3
+      with:
+        path: some/other/path
+
+    - name: Verify successful download
+      run: |
+        $fileA = "${{ github.workspace }}/some/other/path/Artifact-A-${{ matrix.runs-on }}/file-A.txt"
+        $fileB = "${{ github.workspace }}/some/other/path/Artifact-B-${{ matrix.runs-on }}/file-B.txt"
+        if(!(Test-Path -path $fileA) -or !(Test-Path -path $fileB))
+        {
+            Write-Error "Expected files do not exist"
+        }
+        if(!((Get-Content $fileA) -ceq "Lorem ipsum dolor sit amet") -or !((Get-Content $fileB) -ceq "Hello world from file B"))
+        {
+            Write-Error "File contents of downloaded artifacts are incorrect"
+        }
+      shell: pwsh
```

Comment:
Why do we need this?

Reply:
Same as above. Is part of your original unit tests. I am only fixing the names to avoid errors due to conflicts with each of the runs.